js/searchResults.js

```diff
@@ -0,0 +1,74 @@
+const userInput = document.querySelector(".searchInputBox");
+const searchBtn = document.querySelector(".btn");
+const list = document.querySelector(".resultList");
+const loadingSpinner = document.querySelector(".loadingSpinner");
+
+class ResultListItem {
+    constructor(tag) {
+        this.tag = tag
+        this.elem = this.createElement(tag);
+    }
+
+    createElement(tag) {
+        const elem = document.createElement(tag);
+        return elem;
+    }
+
+    appendTo(selector) {
+        const wrap = document.querySelector(selector);
+        wrap.append(this.elem);
+    }
+
+    write(text, query) {
+        const highlightedText = text.replace(new RegExp(query, "gi"), match => `<mark>${match}</mark>`);
+        this.elem.innerHTML = highlightedText;
+    }
+
+    addClass(...classNames) {
+        this.elem.classList.add(...classNames);
+    }
+}
+
+class SearchResult {
+    constructor(selector) {
+        this.wrapper = selector;
+        this.getSearchList();
+    }
+
+    getSearchList() {
```

Comment:
you should seperate the code that fetchs the data from the code that renders it do the DOM into 2 different function. the Rendering function should be inside this class, and the fetching function should be inside searchForm class.